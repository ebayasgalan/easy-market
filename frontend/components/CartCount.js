import React, { Component } from "react";
import PropTypes from "prop-types";
import { TransitionGroup, CSSTransition } from "react-transition-group";
import styled from "styled-components";

const AnimationStyles = styled.span`
  position: relative;
  .count {
    display: block;
    position: relative;
    transition: all 0.3s;
    backface-visibility: hidden;
  }
  .count-enter {
    transform: rotateX(0.5turn);
  }
  .count-enter-active {
    transform: rotateX(0);
  }
  .count-exit {
    top: 0;
    position: absolute;
    transform: rotateX(0);
  }
  .count-exit-active {
    transform: rotateX(0.5turn);
  }
`;

const Dot = styled.div`
  color: white;
  background: ${props => props.theme.blue};
  border-radius: 50%;
  padding: 0.5rem;
  line-height: 2rem;
  min-width: 3rem;
  margin-left: 1rem;
  font-weight: 100;
  font-feature-settings: "tnum";
  font-variant-numeric: tabular-nums;
`;

const CartCount = ({ count }) => (
  <AnimationStyles>
    <TransitionGroup>
      <CSSTransition
        unmountOnExit
        className="count"
        classNames="count"
        key={count}
        timeout={{ enter: 300, exit: 300 }}
      >
        <Dot>{count}</Dot>
      </CSSTransition>
    </TransitionGroup>
  </AnimationStyles>
);

export default CartCount;
