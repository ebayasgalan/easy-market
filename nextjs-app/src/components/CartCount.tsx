'use client';

import styled from 'styled-components';

const Count = styled.div`
  background: var(--red);
  color: white;
  border-radius: 50%;
  width: 25px;
  text-align: center;
  margin: auto;
  font-weight: 600;
  font-size: 15px;
`;

export default function CartCount({ count }) {
  return (
    <Count>
      {count}
    </Count>
  );
}