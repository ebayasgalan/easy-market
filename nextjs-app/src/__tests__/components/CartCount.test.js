import { render, screen } from '@testing-library/react';
import CartCount from '../../components/CartCount';

describe('<CartCount/>', () => {
  it('Renders', () => {
    render(<CartCount count={10} />);
  });
  it('Matches snapshot', () => {
    const { container } = render(<CartCount count={12} />);
    expect(container).toMatchSnapshot();
  });
  it('updates via props', async () => {
    const { container, rerender, debug } = render(<CartCount count={11} />);
    expect(container.textContent).toBe('11');
    // Update the props
    rerender(<CartCount count="12" />);
    expect(container.textContent).toBe('12');
    // expect(container).toMatchSnapshot();
  });
});