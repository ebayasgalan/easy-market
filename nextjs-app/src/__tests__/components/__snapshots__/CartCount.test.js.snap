// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CartCount/> Matches snapshot 1`] = `
<div>
  <div
    class="count"
  >
    12
  </div>
</div>
`;

exports[`<CartCount/> updates via props 1`] = `
<div>
  <div
    class="count"
  >
    12
  </div>
</div>
`;
